### PR TITLE
tools/networking: replace name with pname&version

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#103997

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 132341` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages marked as broken and skipped:</summary>
  <ul>
    <li>aqemu</li>
    <li>hhvm</li>
    <li>libsForQt512.kcachegrind</li>
    <li>libsForQt514.kcachegrind</li>
    <li>linuxPackages-libre.virtualbox</li>
    <li>linuxPackages_hardkernel_4_14.virtualbox</li>
    <li>linuxPackages_hardkernel_latest.virtualbox</li>
    <li>linuxPackages_latest-libre.virtualbox</li>
    <li>multibootusb</li>
    <li>php74Packages.phpmd</li>
    <li>php80Packages.phpmd</li>
    <li>vtun</li>
  </ul>
</details>
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixosTest-test</li>
    <li>tests.trivial</li>
  </ul>
</details>
<details>
  <summary>22 packages failed to build:</summary>
  <ul>
    <li>cloud-init</li>
    <li>drush</li>
    <li>linuxPackages.virtualbox (linuxPackages_5_10.virtualbox)</li>
    <li>linuxPackages_4_14.virtualbox</li>
    <li>linuxPackages_4_19.virtualbox</li>
    <li>linuxPackages_4_4.virtualbox</li>
    <li>linuxPackages_4_9.virtualbox</li>
    <li>linuxPackages_5_12.virtualbox</li>
    <li>linuxPackages_5_13.virtualbox (linuxPackages_latest.virtualbox)</li>
    <li>linuxPackages_5_4.virtualbox</li>
    <li>linuxPackages_hardened.virtualbox</li>
    <li>linuxPackages_latest_hardened.virtualbox</li>
    <li>linuxPackages_lqx.virtualbox</li>
    <li>linuxPackages_testing_bcachefs.virtualbox</li>
    <li>linuxPackages_xanmod.virtualbox</li>
    <li>linuxPackages_zen.virtualbox</li>
    <li>open-watcom-bin</li>
    <li>prayer</li>
    <li>virtualbox</li>
    <li>virtualboxHardened</li>
    <li>virtualboxHeadless</li>
    <li>virtualboxWithExtpack</li>
  </ul>
</details>
<details>
  <summary>94 packages built:</summary>
  <ul>
    <li>adminer</li>
    <li>alpine-make-vm-image</li>
    <li>php (apacheHttpdPackages.php ,php80)</li>
    <li>arcanist</li>
    <li>azureus</li>
    <li>bookstack</li>
    <li>cloud-utils</li>
    <li>connman</li>
    <li>connman-gtk</li>
    <li>connman-ncurses</li>
    <li>connmanFull</li>
    <li>connman_dmenu</li>
    <li>coretoppings</li>
    <li>cot (python39Packages.cot)</li>
    <li>diffoscope</li>
    <li>easyeffects</li>
    <li>gnome.gnome-boxes</li>
    <li>networkmanager-vpnc (gnome.networkmanager-vpnc)</li>
    <li>icingaweb2</li>
    <li>libguestfs</li>
    <li>libsForQt5.kcachegrind (libsForQt515.kcachegrind ,plasma5Packages.kcachegrind)</li>
    <li>lima</li>
    <li>lsp-plugins</li>
    <li>matomo</li>
    <li>matomo-beta</li>
    <li>nagios</li>
    <li>nextcloud-news-updater</li>
    <li>noip</li>
    <li>ntopng</li>
    <li>nylon</li>
    <li>out-of-tree</li>
    <li>pcapfix</li>
    <li>pdnsd</li>
    <li>pdsh</li>
    <li>phoronix-test-suite</li>
    <li>php74</li>
    <li>php74Extensions.imap</li>
    <li>php74Packages.box</li>
    <li>php74Packages.composer</li>
    <li>php74Packages.deployer</li>
    <li>php74Packages.php-cs-fixer</li>
    <li>php74Packages.php-parallel-lint</li>
    <li>php74Packages.phpcbf</li>
    <li>php74Packages.phpcs</li>
    <li>php74Packages.phpstan</li>
    <li>php74Packages.psalm</li>
    <li>php74Packages.psysh</li>
    <li>php80Extensions.imap</li>
    <li>php80Packages.box</li>
    <li>php80Packages.composer</li>
    <li>php80Packages.deployer</li>
    <li>php80Packages.php-cs-fixer</li>
    <li>php80Packages.php-parallel-lint</li>
    <li>php80Packages.phpcbf</li>
    <li>php80Packages.phpcs</li>
    <li>php80Packages.phpstan</li>
    <li>php80Packages.psalm</li>
    <li>php80Packages.psysh</li>
    <li>polygraph</li>
    <li>pulseeffects-legacy</li>
    <li>python38Packages.cot</li>
    <li>python38Packages.guestfs</li>
    <li>python39Packages.guestfs</li>
    <li>qcachegrind</li>
    <li>qemu</li>
    <li>qemu-utils</li>
    <li>qemu_full</li>
    <li>qemu_kvm</li>
    <li>qemu_test</li>
    <li>qemu_xen (qemu_xen_4_10)</li>
    <li>qemu_xen-light (qemu_xen_4_10-light)</li>
    <li>qtemu</li>
    <li>simh</li>
    <li>solo5</li>
    <li>srelay</li>
    <li>ssmtp</li>
    <li>surfraw</li>
    <li>swec</li>
    <li>toxvpn</li>
    <li>trickle</li>
    <li>ucspi-tcp</li>
    <li>udptunnel</li>
    <li>unit</li>
    <li>uwimap</li>
    <li>vagrant</li>
    <li>vde2</li>
    <li>vlan</li>
    <li>vpnc</li>
    <li>wakelan</li>
    <li>webalizer</li>
    <li>wp-cli</li>
    <li>x11_ssh_askpass</li>
    <li>xnbd</li>
    <li>yle-dl</li>
  </ul>
</details>

File: pkgs/tools/networking/vde2/default.nix [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 
 stdenv.mkDerivation rec {
-  name = "vde2-2.3.2";
+  pname = "vde2";
+  version = "2.3.2";
 
   src = fetchurl {
-    url = "mirror://sourceforge/vde/vde2/2.3.1/${name}.tar.gz";
+    url = "mirror://sourceforge/vde/vde2/${version}/vde2-${version}.tar.gz";
     sha256 = "14xga0ib6p1wrv3hkl4sa89yzjxv7f1vfqaxsch87j6scdm59pr2";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/vde2/default.nix:9]: I think the hash might be incorrect here.

Comment (Stunkymonkey) [reply pkgs/tools/networking/vde2/default.nix:9]: i tested it and the hash is correct

**Comment by davidak** [comment]
The failed builds seem to mostly be also broken in master, like drush and prayer. virtualbox is not broken.

```
[davidak@gaming:~/code/nixpkgs]$ nix run nixpkgs.hydra-check -c hydra-check drush
[16 copied (56.7 MiB), 12.1 MiB DL]
Build Status for nixpkgs.drush.x86_64-linux on unstable
✖ (Dependency failed) drush-6.1.0 from 2021-07-29 - https://hydra.nixos.org/build/149008609

Last Builds:
✖ (Dependency failed) drush-6.1.0 from 2021-07-23 - https://hydra.nixos.org/build/148445739
✖ (Dependency failed) drush-6.1.0 from 2021-07-21 - https://hydra.nixos.org/build/148277736
✖ (Dependency failed) drush-6.1.0 from 2021-07-20 - https://hydra.nixos.org/build/148087103
✔ drush-6.1.0 from 2021-07-06 - https://hydra.nixos.org/build/147217399
✔ drush-6.1.0 from 2021-07-02 - https://hydra.nixos.org/build/146832409
✔ drush-6.1.0 from 2021-06-26 - https://hydra.nixos.org/build/146494290
```

@Stunkymonkey can you rebase and fix the merge conflict? I think we can merge it then.

**Comment by SuperSandro2000** [comment]
@Stunkymonkey please resolve the merge conflicts...

nvm

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 132341` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vde2</li>
  </ul>
</details>

**Pull request merged**